src/slam_karto.cpp

```diff
@@ -291,12 +345,19 @@ SlamKarto::~SlamKarto()
     transform_thread_->join();
     delete transform_thread_;
   }
+  if(vis_thread_)
+  {
+    vis_thread_->join();
+    delete vis_thread_;
+  }
+  if(solver_)
```

Comment:
I restored these delete statements : if we remove them, we systematically get the following warning when the slam_karto node is terminated : 
```
^C[slam_karto-2] killing on exit
Warning: class_loader.ClassLoader: SEVERE WARNING!!! Attempting to unload library while objects created by this loader exist in the heap! You should delete your objects before attempting to unload the library or destroying the ClassLoader. The library will NOT be unloaded.
         at line 94 in /tmp/binarydeb/ros-indigo-class-loader-0.3.4/src/class_loader.cpp
```
cf section 1.5 of http://wiki.ros.org/pluginlib/Troubleshooting
which seems to happen because the class loader instances `solver_loader_` and `visualizer_loader_` get deleted before the shared ptrs. So one way to enforce the order of deletion is to use these delete statements.
Calling `reset()` does seem to work. I believe it's because we have multiple references to the pointers and reset will only decrement the ref counter by one.
I am open to other suggestions.